### PR TITLE
Switch to goldmark and glamour for Markdown-to-package-description rendering

### PR DESCRIPTION
**Pull request by anthonyfok**
This pull request switches the current blackfriday-based renderer to goldmark-based [glamour](https://github.com/charmbracelet/glamour) Markdown renderer for ANSI terminal output and supports plain-text output too.

The goal is to have README.md more properly formatted into our needed package description, thus requiring less manual editing.

The caveat is that the resulting dh-make-golang binary does grow somewhat bigger, from about 6.5 MB to 17 MB, though I think the use of glamour and friends allow us to improve the package description formatting with minimal amount of work.

I'd like to solicit feedback before merging this PR.  All comments are welcome!

***

## Example 1: Before

```
 Glamour<p>
     <img src="https://stuff.charm.sh/glamour/glamour-github-header.png"
     width="245" alt="Glamour Title Treatment"><br> <a
     href="https://github.com/charmbracelet/glamour/releases"><img
     src="https://img.shields.io/github/release/charmbracelet/glamour.svg"
     alt="Latest Release"></a> <a
     href="https://pkg.go.dev/github.com/charmbracelet/glamour?tab=doc"><img
     src="https://godoc.org/github.com/golang/gddo?status.svg"
     alt="GoDoc"></a> <a
     href="https://github.com/charmbracelet/glamour/actions"><img
     src="https://github.com/charmbracelet/glamour/workflows/build/badge.svg"
     alt="Build Status"></a> <a
     href="https://coveralls.io/github/charmbracelet/glamour?branch=master"><img
     src="https://coveralls.io/repos/github/charmbracelet/glamour/badge.svg?branch=master"
     alt="Coverage Status"></a> <a
     href="https://goreportcard.com/report/charmbracelet/glamour"><img
     src="https://goreportcard.com/badge/charmbracelet/glamour" alt="Go
     ReportCard"></a>
 </p> Stylesheet-based markdown rendering for your CLI apps.
 .
 Glamour dark style example
 .
 glamour lets you render markdown
 (https://en.wikipedia.org/wiki/Markdown) documents & templates on ANSI
 (https://en.wikipedia.org/wiki/ANSI_escape_code) compatible terminals. You
 can create your own stylesheet or simply use one of the stylish defaults.
 Usage ```go import "github.com/charmbracelet/glamour"
 .
 in := `# Hello World
 .
 This is a simple example of Markdown rendering
 with Glamour!  Check out the other examples
 (https://github.com/charmbracelet/glamour/tree/master/examples) too.
 .
 Bye!  `
 .
 out, err := glamour.Render(in, "dark") fmt.Print(out) ```
 .
 .
 Custom Renderer ```go import "github.com/charmbracelet/glamour"
 .
 r, _ := glamour.NewTermRenderer(
     // detect background color and pick either the default dark or
     light theme glamour.WithAutoStyle(), // wrap output at specific
     width glamour.WithWordWrap(40),
 )
 .
 out, err := r.Render(in) fmt.Print(out) ``` Styles
 You can find all available default styles in our gallery
 (https://github.com/charmbracelet/glamour/tree/master/styles/gallery).
 Want to create your own style? Learn
 how! (https://github.com/charmbracelet/glamour/tree/master/styles)
 .
 There are a few options for using a custom style: 1. Call
 glamour.Render(inputText, "desiredStyle") 1. Set the GLAMOUR_STYLE
 environment variable to your desired default style or a file location
 for a style and call glamour.RenderWithEnvironmentConfig(inputText)
 1. Set the GLAMOUR_STYLE environment variable and pass
 glamour.WithEnvironmentConfig() to your custom renderer Glamourous
 Projects Check out these projects, which use glamour: - Glow
 (https://github.com/charmbracelet/glow), a markdown renderer for the
 command-line.  - GitHub CLI (https://github.com/cli/cli), GitHub’s
 official command line tool.  - GLab (https://github.com/profclems/glab),
 An open source GitLab command line tool.  License MIT
 (https://github.com/charmbracelet/glamour/raw/master/LICENSE)
 --------------------------------------------------------------------------------
 .
 Part of Charm (https://charm.sh).
 .
 .
 .
 Charm热爱开源 • Charm loves open source
```

## Example 1: After

```
 Glamour
 .
 .
 Stylesheet-based markdown rendering for your CLI apps.
 .
 [Image: Glamour dark style example]
 (https://stuff.charm.sh/glamour/glamour-example.png)
 .
 `glamour` lets you render markdown
 (https://en.wikipedia.org/wiki/Markdown) documents & templates on ANSI
 (https://en.wikipedia.org/wiki/ANSI_escape_code) compatible terminals.
 You can create your own stylesheet or simply use one of the stylish
 defaults.
 .
 Usage
 .
   import "github.com/charmbracelet/glamour"
 .
   in := `# Hello World
 .
   This is a simple example of Markdown rendering with Glamour!
   Check out the [other
 examples](https://github.com/charmbracelet/glamour/tree/master/examples)
 too.
 .
   Bye!
   `
 .
   out, err := glamour.Render(in, "dark")
   fmt.Print(out)
 .
 .
 Custom Renderer
 .
   import "github.com/charmbracelet/glamour"
 .
   r, _ := glamour.NewTermRenderer(
       // detect background color and pick either the default dark or
 light theme
       glamour.WithAutoStyle(),
       // wrap output at specific width
       glamour.WithWordWrap(40),
   )
 .
   out, err := r.Render(in)
   fmt.Print(out)
 .
 Styles
 .
 You can find all available default styles in our gallery
 (https://github.com/charmbracelet/glamour/tree/master/styles/gallery).
 Want to create your own style? Learn how!
 (https://github.com/charmbracelet/glamour/tree/master/styles)
 .
 There are a few options for using a custom style:
 .
  1. Call `glamour.Render(inputText, "desiredStyle")`
  2. Set the `GLAMOUR_STYLE` environment variable to your desired default
     style or a file location for a style and call
     `glamour.RenderWithEnvironmentConfig(inputText)`
  3. Set the `GLAMOUR_STYLE` environment variable and pass
     `glamour.WithEnvironmentConfig()` to your custom renderer
 .
 Glamourous Projects
 .
 Check out these projects, which use `glamour`:
 .
  * Glow (https://github.com/charmbracelet/glow), a markdown renderer for
    the command-line.
  * GitHub CLI (https://github.com/cli/cli), GitHub’s official command
    line tool.
  * GLab (https://github.com/profclems/glab), An open source GitLab
    command line tool.
 .
 License
 .
 MIT (https://github.com/charmbracelet/glamour/raw/master/LICENSE)
 .
 ------------------------------------------------------------------------
 .
 Part of Charm (https://charm.sh).
 .
 .
 Charm热爱开源 • Charm loves open source
```

***

## Example 2: Before

```
 GitHub CLI gh is GitHub on the command line. It brings pull requests,
 issues, and other GitHub concepts to the terminal next to where you are
 already working with git and your code.
 .
 screenshot of gh pr status
 .
 GitHub CLI is available for repositories hosted on GitHub.com and GitHub
 Enterprise Server 2.20+, and to install on macOS, Windows, and Linux.
 Documentation See the manual (https://cli.github.com/manual/) for
 setup and usage instructions.  Contributing If anything feels off,
 or if you feel that some functionality is missing, please check out
 the contributing page (./.github/CONTRIBUTING.md). There you will find
 instructions for sharing your feedback, building the tool locally, and
 submitting pull requests to the project.  <!-- this anchor is linked to
 from elsewhere, so avoid renaming it -->InstallationmacOS gh is available
 via Homebrew (https://brew.sh), MacPorts (https://www.macports.org),
 Conda (https://docs.conda.io/en/latest/), and as a downloadable binary
 from the releases page (https://github.com/cli/cli/releases/latest).
 Homebrew | Install:          | Upgrade:          | | -----------------
 | ----------------- | | brew install gh | brew upgrade gh | MacPorts |
 Install:               | Upgrade:                                       |
 | ---------------------- | ----------------------------------------------
 | | sudo port install gh | sudo port selfupdate && sudo port
 upgrade gh | Conda | Install:                                 |
 Upgrade:                                |
 |------------------------------------------|-----------------------------------------|
 | conda install gh --channel conda-forge | conda update gh --channel
 conda-forge |
```

## Example 2: After

```
 GitHub CLI
 .
 `gh` is GitHub on the command line. It brings pull requests, issues, and
 other GitHub concepts to the terminal next to where you are already
 working with `git` and your code.
 .
 [Image: screenshot of gh pr status] (https://user-
 images.githubusercontent.com/98482/84171218-327e7a80-aa40-11ea-8cd1-
 5177fc2d0e72.png)
 .
 GitHub CLI is available for repositories hosted on GitHub.com and GitHub
 Enterprise Server 2.20+, and to install on macOS, Windows, and Linux.
 .
 Documentation
 .
 See the manual (https://cli.github.com/manual/) for setup and usage
 instructions.
 .
 Contributing
 .
 If anything feels off, or if you feel that some functionality is
 missing, please check out the contributing page
 (/.github/CONTRIBUTING.md). There you will find instructions for sharing
 your feedback, building the tool locally, and submitting pull requests
 to the project.
 .
 .
 Installation
 .
 macOS
 .
 `gh` is available via Homebrew (https://brew.sh), MacPorts
 (https://www.macports.org), Conda (https://docs.conda.io/en/latest/),
 and as a downloadable binary from the releases page
 (https://github.com/cli/cli/releases/latest).
 .
 Homebrew
 .
      INSTALL:     |    UPGRADE:
 ------------------+------------------
   brew install gh | brew upgrade gh
 .
 MacPorts
 .
         INSTALL:       |            UPGRADE:
 -----------------------+---------------------------------
   sudo port install gh | sudo port selfupdate && sudo
                        | port upgrade gh
 .
 Conda
 .
              INSTALL:            |            UPGRADE:
 ---------------------------------+---------------------------------
   conda install gh --channel     | conda update gh --channel
   conda-forge                    | conda-forge
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,13 @@
 module github.com/Debian/dh-make-golang
 
-go 1.13
+go 1.16
 
 require (
+	github.com/charmbracelet/glamour v0.3.0
```
Comment (creekorful) [inline go.mod:6]: Not already in Debian as I can see, we will need to package it

File: description.go [hunk 2]
```diff
@@ -1,21 +1,45 @@
...
+		}
+		if line == "" {
+			line = "."
+			prefix = ""
+			next_prefix = ""
+		}
+		output += " " + prefix + line + "\n"
 	}
-	return strings.Replace(raw, "\n", "\n ", -1)
+	return output
```
Comment (creekorful) [inline description.go:42]: The processing looks really nice! But I think adding a little unit test for all the prefix, next_prefix could be really great in helping further regression. What do you think?

**Review by creekorful — COMMENTED** [review]
The improvement is quite interesting, I really like how the description are rendered now

File: go.mod (hunk 1 above)
Comment (anthonyfok) [reply go.mod:6]: Yay!  The Debian FTP masters approved the package, and it is now in unstable, and hopefully in testing soon: https://tracker.debian.org/pkg/golang-github-charmbracelet-glamour

File: description.go (hunk 2 above)
Comment (anthonyfok) [reply description.go:42]: That's a great idea!  Thank you!  Will be learning how to do it soon  (I'm still pretty new to test-driven development :sweat_smile:)

**Pull request merged**